includes/services/extended/linkedin.php

```diff
@@ -1,97 +1,153 @@
 <?php
 
 /**
- * LinkedIn service definition for Keyring. Clean implementation of OAuth1
+ * LinkedIn service definition for Keyring. Clean implementation of OAuth2
  */
 
-class Keyring_Service_LinkedIn extends Keyring_Service_OAuth1 {
+class Keyring_Service_LinkedIn extends Keyring_Service_OAuth2 {
 	const NAME  = 'linkedin';
 	const LABEL = 'LinkedIn';
 
+	var $person = array();
+
 	function __construct() {
 		parent::__construct();
 
-		$this->authorization_header = true;
-		$this->authorization_realm = "api.linkedin.com";
-
 		// Enable "basic" UI for entering key/secret
 		if ( ! KEYRING__HEADLESS_MODE ) {
 			add_action( 'keyring_linkedin_manage_ui', array( $this, 'basic_ui' ) );
 			add_filter( 'keyring_linkedin_basic_ui_intro', array( $this, 'basic_ui_intro' ) );
 		}
 
-		$this->set_endpoint( 'request_token', 'https://api.linkedin.com/uas/oauth/requestToken', 'POST' );
-		$this->set_endpoint( 'authorize',     'https://api.linkedin.com/uas/oauth/authenticate', 'GET'  );
-		$this->set_endpoint( 'access_token',  'https://api.linkedin.com/uas/oauth/accessToken',  'GET'  );
+		$this->set_endpoint( 'authorize',    'https://www.linkedin.com/oauth/v2/authorization', 'GET' );
+		$this->set_endpoint( 'access_token', 'https://www.linkedin.com/oauth/v2/accessToken',   'POST' );
+		$this->set_endpoint( 'self',         'https://api.linkedin.com/v2/me',            'GET' );
+		$this->set_endpoint( 'profile_pic',  'https://api.linkedin.com/v2/me/picture-urls::(original)/', 'GET' );
```

Comment:
Is this misalignment just something wonky in how GH is displaying it? If not let's line these params in this block up.